### PR TITLE
VEM support in FEM kernel interface

### PR DESCRIPTION
**Pull request by andrea-borio**
This pull request provides the features needed to use Virtual Element Methods in Finite Element kernel interfaces. It also modifies `LaplaceFEMKernels` in order to use either VEM or FEM to solve the Laplace equation.

### Summary of changes
- `FiniteElementDiscretization::factory` creates instances of `ConformingVirtualElementOrder1` if the flag `useVirtualElements` is defined and set to one in the input xml file;
- `FiniteElementDispatch::dispatch3D` tries to apply the input lambda to VEM classes if the macro `GEOSX_DISPATCH_VEM` is defined (see [Notes](#Notes) below);
- `finiteElement::KernelBase` and `finiteElement::ImplicitKernelBase` are modified with the new syntax implementing a common FEM and VEM interface, introduced in #1489. In particular, a `FE_TYPE::MeshData` type struct is added to the class and a `FE_TYPE::StackVariables` type struct is added to the kernel's `StackVariables`;
- `ProblemManager` modified with the new syntax to correctly compute VEM shape function gradients when `FiniteElementDiscretization` finds the flag `useVirtualElements` in the input xml file;
- `LaplaceFEMKernels` modified with the new common FEM+VEM syntax.

### Notes
If a class wants to call `FiniteElementDispatch::dispatch3D` supporting VEM it has to `#define GEOSX_DISPATCH_VEM` before including `finiteElement/FiniteElementDispatch.hpp`. This is currently done in `mainInterface/ProblemManager.cpp` and `physicsSolvers/simplePDE/LaplaceFEMKernels.hpp`. This is a _temporary_ workaround to avoid some CUDA `Formal parameter space overflowed` build errors that are issued when MultiphasePoromechanics kernels are compiled with VEM.

Rebaseline done in https://github.com/GEOSX/integratedTests/pull/192

### CODE REVIEW

File: src/coreComponents/finiteElement/elementFormulations/H1_Tetrahedron_Lagrange1_Gauss1.hpp [hunk 1]
```diff
@@ -238,20 +246,22 @@ class H1_Tetrahedron_Lagrange1_Gauss1 final : public FiniteElementBase
...
+template< typename SUBREGION_TYPE >
 GEOSX_FORCE_INLINE
 void H1_Tetrahedron_Lagrange1_Gauss1::
   fillMeshData( NodeManager const & GEOSX_UNUSED_PARAM( nodeManager ),
                 EdgeManager const & GEOSX_UNUSED_PARAM( edgeManager ),
                 FaceManager const & GEOSX_UNUSED_PARAM( faceManager ),
-                CellElementSubRegion const & GEOSX_UNUSED_PARAM( cellSubRegion ),
-                MeshData & GEOSX_UNUSED_PARAM( initialization ) )
+                SUBREGION_TYPE const & GEOSX_UNUSED_PARAM( cellSubRegion ),
+                MeshData< SUBREGION_TYPE > & GEOSX_UNUSED_PARAM( initialization ) )
```
Comment (castelletto1) [inline src/coreComponents/finiteElement/elementFormulations/H1_Tetrahedron_Lagrange1_Gauss1.hpp:256]: @francoishamon @rrsettgast @joshua-white What was the final decision about using `GEOSX_UNUSED_PARAM`? I vaguely recall that at some point we were considering removing it from the function declaration, and used instead `GEOSX_UNUSED_VAR` in the function body.

**Review by francoishamon — APPROVED** [review]
Great work, thanks @andrea-borio

**Comment by TotoGaz** [comment]
@francoishamon Didn't you remove some `*Laplace*.xml` files that were pointed by links from `integratedTests`? Links are dead now.

**Comment by francoishamon** [comment]
@andrea-borio You did not do anything wrong, the dead links mentioned by @TotoGaz were my fault. 

I had to create a new branch in `integratedTests` for your baselines because I noticed that on your `feature/borio/laplaceFEMTests` the baselines had been "corrupted" by the successive merges with `develop` (unfortunately, we cannot merge baselines). In the process of regenerating the baselines in the new branch, I forgot to delete the symbolic links. This is fixed now.

**Pull request merged**